EIPS/eip-9999.md

```diff
@@ -0,0 +1,447 @@
+MAX_TRANSACTIONS_PER_CHUNK = 2**10 # Maximum number of transactions in a chunk (1,024 tx per chunk)
+```
+
+### Data Structures
+
+#### Execution Layer Structures
+
+```python
+# Chunk header containing execution commitments
+class ExecutionChunkHeader(Container):
+    slot: Slot                       # The beacon chain slot
+    index: ChunkIndex                # Position in block (0 to MAX_CHUNKS_PER_BLOCK-1)
+    is_last: bool                    # Whether chunk is the last chunk in block
+    chunk_access_list_hash: Root     # The hash of CAL (keccak-256 of the RLP-encoding)
+    pre_chunk_tx_count: uint32       # Number of transactions in a block before the chunk
+    pre_chunk_gas_used: uint64       # Gas consumed in block before the chunk
+    pre_chunk_blob_gas_used: uint64  # Blob gas consumed in block before the chunk
+    txs_root: Root                   # Merkle root of transactions
+    gas_used: uint64                 # Gas consumed in chunk
+    blob_gas_used: uint64            # Blob gas consumed in chunk
+    withdrawals_root: Root           # Merkle root of withdrawals (only in last chunk)
+
+# Self-contained execution chunk
+class ExecutionChunk(Container):
+    chunk_header: ExecutionChunkHeader
+    transactions: List[Transaction, MAX_TRANSACTIONS_PER_CHUNK]
+    withdrawals: List[Withdrawal, MAX_WITHDRAWALS_PER_PAYLOAD]  # Only in last chunk
+```
+
+#### Chunk Access Lists (CALs)
+
+CALs use the same RLP encoding as Block Access Lists from [EIP-7928](./eip-7928.md) but scoped to individual chunks. They contain state diffs from chunk execution:
+
+```python
+# Type aliases matching EIP-7928
+Address = bytes  # 20-byte Ethereum address
+StorageKey = bytes  # 32-byte storage slot key  
+StorageValue = bytes  # 32-byte storage value
+Balance = uint256  # Post-execution balance
+Nonce = uint64  # Account nonce
+CodeData = bytes  # Contract bytecode
+TxIndex = uint16  # Transaction index within block
+
+# Storage change: [tx_index, new_value]
+StorageChange = [TxIndex, StorageValue]
+
+# Balance change: [tx_index, post_balance]
+BalanceChange = [TxIndex, Balance]
+
+# Nonce change: [tx_index, new_nonce]
+NonceChange = [TxIndex, Nonce]
+
+# Code change: [tx_index, new_code]
+CodeChange = [TxIndex, CodeData]
+
+# All changes to a single storage slot
+SlotChanges = [StorageKey, List[StorageChange]]
+
+# All changes for a single account within the chunk
+AccountChanges = [
+    Address,                    # address
+    List[SlotChanges],         # storage_changes
+    List[StorageKey],          # storage_reads (read-only)
+    List[BalanceChange],       # balance_changes
+    List[NonceChange],         # nonce_changes
+    List[CodeChange]           # code_changes
+]
+
+# Chunk Access List: All account changes in the chunk
+ChunkAccessList = List[AccountChanges]
+```
+
+CALs are incremental. This means that changes and storage reads from earlier chunks are not included in CALs of later chunks.
```

Comment:
So a CAL will only have the access lists for the tx indices relevant to that CAL? With the first CAL having system access list also, and the final CAL also has the system operations done after all txs (if any?)